forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt

```diff
@@ -62,10 +63,36 @@ interface Authenticator {
         signedJWT.sign(RSASSASigner(keyPair.privateKey))
         return signedJWT.serialize()
     }
+
+    /**
+     * sign the challenge sent from the server and generate signed JWT
+     * @param userKey User Information
+     * @param challenge challenge received from server
+     * @param privateKey private key for the given alias
+     */
+    fun sign(userKey: UserKey, challenge: String, privateKey: PrivateKey? = getPrivateKey(userKey.keyAlias)): String {
+        val signedJWT = SignedJWT(
+            JWSHeader.Builder(JWSAlgorithm.RS512)
+                .keyID(userKey.kid)
+                .build(), JWTClaimsSet.Builder()
+                .subject(userKey.userId)
+                .claim("challenge", challenge)
+                .build()
+        )
+        signedJWT.sign(RSASSASigner(privateKey))
+        return signedJWT.serialize()
+    }
     /**
      * check biometric is supported
      */
     fun isSupported(): Boolean
+
+    /**
+     * Get the private key for the user
+     * @param keyAlias The user keyAlias
+     */
+    fun getPrivateKey(keyAlias: String): PrivateKey?
```

Comment:
This reads a bit scary. Is this inaccessible from outside this context because the function itself is defined on an `internal class`? Sorry if this is a silly question, but `getPrivateKey` just sounds worrying :)

Reply:
its a wonderful question. i am addressing this shortly